test/Microsoft.Azure.Devices.Edge.Test.Common/config/BaseModuleConfigBuilder.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common.Config
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public abstract class BaseModuleConfigBuilder : IModuleConfigBuilder
+    {
+        IDictionary<string, object> DesiredProperties { get; }
+
+        IDictionary<string, string> Env { get; }
+
+        protected IDictionary<string, object> Deployment { get; }
+
+        protected IDictionary<string, object> Settings { get; }
+
+        public string Name { get; }
+
+        protected BaseModuleConfigBuilder(string name, string image)
+        {
+            this.Deployment = new Dictionary<string, object>()
+            {
+                ["type"] = "docker"
+            };
+            this.DesiredProperties = new Dictionary<string, object>();
+            this.Env = new Dictionary<string, string>();
+            this.Name = name;
+            this.Settings = new Dictionary<string, object>()
```

Comment:
what are possible usage to pass as object type? why not string type?

Reply:
These are designed to be easily convertible to JSON, so the "object" will be a string or a JSON object (another dictionary).